simulator-gui/src/main/scala/it/unibo/scafi/simulation/frontend/model/Node.scala

```diff
@@ -1,13 +1,15 @@
 package it.unibo.scafi.simulation.frontend.model
 
-import it.unibo.scafi.space.Point2D
+import it.unibo.scafi.space.{Point2D, Point3D}
 
 trait Node {
   def id: Int
 
   var export: Any
 
-  var position: Point2D
+  var position: Point3D
+
+  def position2d: Point2D
```

Comment:
Having both fields seem redundant. We should consider dropping `position2d` and just keeping `position`. 2D simulations shouldn't be affected, using `Point3D` objects with `z=0`, right?

Reply:
`position2d` is implemented simply as position with `z=0`. I created it so that the code that expects a `Point2D` still works without making any changes to it. Should i modify each use of `Node`'s position so that they always expect a `Point3D`?

Reply:
We may drop `position2d` and when a 2D position is needed, just provide the 2D-view of the 3D position (maybe an implicit converter can safely help).

Reply:
I'll remove `position2d`. Would it be ok to put `def toPoint2D: Point2D = new Point2D(p.x, p.y)` inside `it.unibo.scafi.space.optimization.RichPoint3D` (or maybe even in `Point3D` itself)? That way everyone can reuse the 2d view logic.

Reply:
I see that
- `Point2D` is a subclass of `Point3D`
- `Point3D` already has an ` implicit def toPoint2D(p: Point3D): Point2D = Point2D(p.x, p.y)` 
The latter which can also be called explicitly, so there is no need of a new method.